lib/date_range.rb

```diff
@@ -0,0 +1,41 @@
+require 'date'
+
+module Hotel
+
+  class DateRange
+    attr_reader :start_date, :end_date
+    def initialize(start_date:, end_date:)
+      @start_date = Date.new(start_date[0],start_date[1],start_date[2])
+      @end_date = Date.new(end_date[0],end_date[1],end_date[2])
+    
+      if @end_date - @start_date <= 0 
+        raise ArgumentError, "#{@end_date} cannot be before or equal to #{@start_date}"
+      end 
+    end 
+
+    #overlap method
+    def overlap(date_range)
+      if @start_date >= date_range.end_date || @end_date <= date_range.start_date
+        return false 
+      else 
+        return true
+      end 
+    end 
```

Comment:
You can directly return the condition for something like this since it's already a boolean value:

```ruby
    def overlap(date_range)
      return !(@start_date >= date_range.end_date || @end_date <= date_range.start_date)
    end 
```

Using [DeMorgan's Law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) you can clean this up even more:


```ruby
    def overlap(date_range)
      return !(@start_date >= date_range.end_date) && @!(end_date <= date_range.start_date)
    end 
```

To eliminate those `!` you can invert the comparisons getting:

```suggestion
    def overlap(date_range)
      return @start_date < date_range.end_date && @end_date > date_range.start_date
    end 
```

Which personally I think reads a lot clearer.